my24.js

```diff
@@ -0,0 +1,138 @@
+class MockServer {
+  #users = [
+    { id: 1, name: "Aaron", age: 20 },
+    { id: 2, name: "Bernard", age: 25 },
+    { id: 3, name: "Connor", age: 36 },
+    { id: 4, name: "Daniel", age: 32 },
+    { id: 5, name: "Evan", age: 30 },
+    { id: 6, name: "Francis", age: 19 },
+    { id: 7, name: "George", age: 21 },
+    { id: 8, name: "Harry", age: 45 },
+    { id: 9, name: "Jack", age: 16 },
+    { id: 10, name: "Kevin", age: 27 },
+    { id: 11, name: "Malcolm", age: 59 },
+  ];
+  result = {
+    status: false,
+    message: "",
+    data: undefined,
+  };
+  isUserExist = (id) =>
+    this.#users.some((currentUser) => {
+      return currentUser.id === id;
+    });
+
+  // get users() {
+  //   return this.#users;
+  // }
+  setResult = (status, message, data) => {
+    this.result.status = status;
+    this.result.message = message;
+    this.result.data = data;
+    return this.result;
+  };
+
+  getAllUsers = () => {
+    this.setResult(true, "is your list of users", this.#users);
+    return this.result;
+  };
+
+  getUser = (id) => {
+    if (this.isUserExist(id)) {
+      const user = this.#users.find((currentUser) => currentUser.id === id);
+      this.setResult(true, "is your user", user);
+    } else {
+      this.setResult(false, "we don't have this user", undefined);
+    }
+    return this.result;
+  };
+
+  addUser = (user) => {
+    const isAllPropExist = !!user.id && !!user.name && !!user.age;
+    if (isAllPropExist && !this.isUserExist(user.id)) {
+      this.#users.push(user);
+      this.setResult(true, "user was added successfully", true);
+    } else {
+      this.setResult(false, "user was't added", false);
+    }
+    return this.result;
+  };
+
+  updateUser = (user) => {
+    if (this.isUserExist(user.id)) {
+      this.#users = this.#users.map((item) =>
+        item.id === user.id ? user : item
```

Comment:
the idea was that id and user should be passed to this function as parameters. The user can contain only those fields that are needed to update, so it would be better to do destructuring here